app/helpers/questions_helper.rb

```diff
@@ -0,0 +1,25 @@
+module QuestionsHelper
+  SORTING_OPTIONS = %w(All Week Month Year).freeze
+
+  def filtering(period)
+    return if period.nil?
+
+    if period == 'All'
+      Rails.cache.fetch("all_stats", expires_in: 1.days) do
+        @asked = Node.questions.to_a.size
```

Comment:
I think if you use `Node.questions.count` it should do a more efficient query, just a SQL SIZE query instead of fetching all the full records and counting them. Maybe same on line below! 👍

Reply:
@jywarren tried `.count` was not working fine  because its  a hash returning  results like so `{82=>1, 85=>2, 86=>2, 87=>1}`. Using `.length` instead. hope that's a better query?